flutter/lib/ui/painting/image_decoding.cc

```diff
@@ -130,12 +129,12 @@ void DecodeImageFromList(Dart_NativeArguments args) {
   PassOwnPtr<DartPersistentValue> callback =
       adoptPtr(new DartPersistentValue(DartState::Current(), callback_handle));
 
-  RefPtr<SharedBuffer> buffer = SharedBuffer::create();
-  buffer->append(reinterpret_cast<const char*>(list.data()),
-                 list.num_elements());
+  const char* bytes = reinterpret_cast<const char*>(list.data());
+  PassOwnPtr<std::vector<char>> buffer =
```

Comment:
> I think a dedicated class that manages resizable allocations would serve us well.

Yeah, you're probably right.

> Generalized lambda capture would avoid the need for the pointer right?

That's right.  A lot of this code becomes prettier with generalized lambda capture.

> Are we switching to C++14? I thought I saw some GN files with that flag.

Fuchsia is on C++14.  The tricky part about switching is getting a toolchain for Android that supports C++14.  It's possible if we upgrade our NDK and bring our own libc++ that we could get C++14 on Android.

We don't have too much of this stuff, so I've just been working around the lack of generalized lambda capture so far.  If it gets too painful, we can try to get a C++14 toolchain going for Android.